target/linux/rockchip/patches-5.10/203-arm64-dts-rockchip-Add-support-for-R2C.patch

```diff
@@ -0,0 +1,73 @@
+From 00c36637656ed89c871495532f4131a1f8864fe5 Mon Sep 17 00:00:00 2001
+From: dEFIX <1132404+911gt3@users.noreply.github.com>
+Date: Tue, 17 Aug 2021 13:53:49 +0800
+Subject: [PATCH 4/4] arm64: dts: rockchip: Add support for R2C
+
+The only difference between R2C and R2S is the Ethernet PHY chip.
+
+Here are the key notes for dts:
+1) rgmii's drive strength should be 8mA
+2) reset-assert-us should be 15000
+3) The fully compatible tx/rx delay is 0x22/0x12
+
+To remove the phy-handle, the following patch is required.
+ ("stmmac: Use phy_find_first() for invalid phy_addr")
+
+Signed-off-by: dEFIX <1132404+911gt3@users.noreply.github.com>
+---
+ .../boot/dts/rockchip/rk3328-nanopi-r2s.dts   | 20 +++++++++++--------
+ 1 file changed, 12 insertions(+), 8 deletions(-)
+
+diff --git a/arch/arm64/boot/dts/rockchip/rk3328-nanopi-r2s.dts b/arch/arm64/boot/dts/rockchip/rk3328-nanopi-r2s.dts
+index 074d45f20342..a3512bf429b7 100644
+--- a/arch/arm64/boot/dts/rockchip/rk3328-nanopi-r2s.dts
++++ b/arch/arm64/boot/dts/rockchip/rk3328-nanopi-r2s.dts
```

Comment:
This alters R2S DTS? Rather create a new one and update the required changes here.

Also the commit message is wrong, as there were more Hardware changes (USB-C being among one of them)

Reply:
I agree that's it better to have a separate R2C, and I'm in the process of making a new DTS and device package for the R2C.

As for the USB-C, I have both an R2S and an R2C, and they both have USB-C.  I've read that an older version of the R2S had a different USB, so it sounds like full support for the R2S would require dealing with both types of USB?  I don't have the older version of the R2S...